### ISSUE TITLE
check_dynamic_table_region_data_validity misses an index equal to the table length

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/nwbinspector#738]
`check_dynamic_table_region_data_validity` flags indices that are greater than the length of the target table:

```python
if np.any(np.asarray(dynamic_table_region.data[:nelems]) > len(dynamic_table_region.table)):
```

Indices are zero-based, so an index equal to the table length is already out of range, and the check lets it through. A DynamicTableRegion pointing at a 10-row table with a value of 10 is reported by HDMF as a warning on read but produces no inspector message. The comparison should be `>=`.

Both the existing test for the "greater than" case and a new test for the boundary value should pass with the change.
